ostd/src/mm/page_table/mod.rs

```diff
@@ -97,87 +99,71 @@ impl PageTable<UserMode> {
             self.root.activate();
         }
     }
+}
 
-    /// Clear the page table.
-    ///
-    /// # Safety
-    ///
-    /// The caller must ensure that:
-    ///  1. No other cursors are accessing the page table.
-    ///  2. No other CPUs activates the page table.
-    pub(in crate::mm) unsafe fn clear(&self) {
-        let mut root_node = self.root.clone_shallow().lock();
-        const NR_PTES_PER_NODE: usize = nr_subpage_per_huge::<PagingConsts>();
-        for i in 0..NR_PTES_PER_NODE / 2 {
-            let root_entry = root_node.entry(i);
-            if !root_entry.is_none() {
-                let old = root_entry.replace(Child::None);
-                // Since no others are accessing the old child, dropping it is fine.
-                drop(old);
+impl PageTable<KernelMode> {
+    /// Create a new kernel page table.
+    pub(crate) fn new_kernel_page_table() -> Self {
+        let kpt = Self::empty();
+
+        // Make shared the page tables mapped by the root table in the kernel space.
+        {
+            const NR_PTES_PER_NODE: usize = nr_subpage_per_huge::<PagingConsts>();
+            let kernel_space_range = NR_PTES_PER_NODE / 2..NR_PTES_PER_NODE;
+
+            let guard = crate::task::disable_preempt();
```

Comment:
```suggestion
            let preempt_guard = disable_preempt();
```

`disable_preempt` is already imported in this PR. There doesn't seem to be a reason to reference it via the full path.

Besides, I think moving the `preempt_guard` closer to `root_node` makes more sense (slightly):
```rust
            let guard = crate::task::disable_preempt();
            let mut root_node = kpt.root.borrow().lock(&guard);

            for i in kernel_space_range {
                // ...
            }
```